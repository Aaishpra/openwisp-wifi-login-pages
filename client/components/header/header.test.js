import {shallow} from "enzyme";
import React from "react";
import {BrowserRouter as Router} from "react-router-dom";
import renderer from "react-test-renderer";

import getConfig from "../../utils/get-config";
import Header from "./header";

const defaultConfig = getConfig("default");
const createTestProps = props => {
  return {
    setLanguage: jest.fn(),
    orgSlug: "default",
    language: "en",
    languages: [{slug: "en", text: "english"}, {slug: "it", text: "italian"}],
    header: defaultConfig.components.header,
    ...props,
  };
};

describe("<Header /> rendering", () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(<Header {...props} />);
  });
  it("should render without links", () => {
    const links = {
      header: {
        ...props.header,
        links: [],
      },
    };
    props = createTestProps(links);
    const component = renderer
      .create(
        <Router>
          <Header {...props} />
        </Router>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
  it("should render with links", () => {
    const component = renderer
      .create(
        <Router>
          <Header {...props} />
        </Router>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
  it("should render 2 links", () => {
    expect(wrapper.find(".owisp-header-desktop-link")).toHaveLength(2);
  });
  it("should render 2 languages", () => {
    expect(wrapper.find(".owisp-header-desktop-language-btn")).toHaveLength(2);
  });
  it("should render english as default language", () => {
    expect(
      wrapper.find(
        ".owisp-header-desktop-language-btn.owisp-header-language-btn-en.active",
      ),
    ).toHaveLength(1);
    expect(
      wrapper.find(
        ".owisp-header-desktop-language-btn.owisp-header-language-btn-it.active",
      ),
    ).toHaveLength(0);
  });
  it("should render logo", () => {
    expect(
      wrapper.find(".owisp-header-logo-image.owisp-header-desktop-logo-image"),
    ).toHaveLength(1);
  });
  it("should not render logo", () => {
    const logo = {
      header: {
        ...props.header,
        logo: null,
      },
    };
    props = createTestProps(logo);
    wrapper = shallow(<Header {...props} />);
    expect(
      wrapper.find(".owisp-header-logo-image.owisp-header-desktop-logo-image"),
    ).toHaveLength(0);
  });
});

describe("<Header /> interactions", () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(<Header {...props} />);
  });
  it("should call setLanguage function when 'language button' is clicked", () => {
    wrapper
      .find(".owisp-header-language-btn-it.owisp-header-desktop-language-btn")
      .simulate("click");
    expect(props.setLanguage).toHaveBeenCalledTimes(1);
  });
});
