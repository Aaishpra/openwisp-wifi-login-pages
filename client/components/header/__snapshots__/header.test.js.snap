// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> rendering should render with links 1`] = `
<div
  className="owisp-header-container"
>
  <div
    className="owisp-header-row-1"
  >
    <div
      className="owisp-header-row-1-inner"
    >
      <div
        className="owisp-header-left"
      >
        <div
          className="owisp-header-logo-div"
        />
      </div>
      <div
        className="owisp-header-right"
      >
        <button
          className="active owisp-header-language-btn owisp-header-language-btn-en"
          onClick={[Function]}
          type="button"
        >
          english
        </button>
        <button
          className="owisp-header-language-btn owisp-header-language-btn-it"
          onClick={[Function]}
          type="button"
        >
          italian
        </button>
      </div>
    </div>
  </div>
  <div
    className="owisp-header-row-2"
  >
    <div
      className="owisp-header-row-2-inner"
    >
      <a
        className="owisp-header-link
                    owisp-header-link-1"
        href="https://link-1.com"
        rel="noreferrer noopener"
        target="_blank"
      >
        link-1
      </a>
      <a
        className="owisp-header-link
                    owisp-header-link-2"
        href="https://link-2.com"
        rel="noreferrer noopener"
        target="_blank"
      >
        link-2
      </a>
    </div>
  </div>
</div>
`;

exports[`<Header /> rendering should render without links 1`] = `
<div
  className="owisp-header-container"
>
  <div
    className="owisp-header-row-1"
  >
    <div
      className="owisp-header-row-1-inner"
    >
      <div
        className="owisp-header-left"
      >
        <div
          className="owisp-header-logo-div"
        />
      </div>
      <div
        className="owisp-header-right"
      >
        <button
          className="active owisp-header-language-btn owisp-header-language-btn-en"
          onClick={[Function]}
          type="button"
        >
          english
        </button>
        <button
          className="owisp-header-language-btn owisp-header-language-btn-it"
          onClick={[Function]}
          type="button"
        >
          italian
        </button>
      </div>
    </div>
  </div>
  <div
    className="owisp-header-row-2"
  >
    <div
      className="owisp-header-row-2-inner"
    />
  </div>
</div>
`;
